Lib/http/server.py

```diff
@@ -731,6 +731,14 @@ def send_head(self):
 
         try:
             fs = os.fstat(f.fileno())
+            # On Windows, _PyTime_gmtime() only allows timestamps larger or
+            # equal to -43200 (1969-12-31 12:00:00). But os.fstat() will
+            # happily return stat_result with st_mtime less then -43200. See
+            # issue #101035. For files with st_mtime less then -43200, clients
+            # might as well get the "updated" version.
+            mtime = fs.st_mtime
+            if sys.platform == 'win32' and mtime < -43200:
+                mtime = time.time()
```

Comment:
Generally a negative value would be due to a zero timestamp on Windows, which is 1601-01-01 00:00:00 (UTC), or -11644473600 as a Unix timestamp. I think a better choice would be `mtime = 0`, the Unix epoch 1970-01-01 00:00:00 (UTC). I'm not concerned with preserving the value of a timestamp between 1601 and 1970, but using the current time loses useful information that the timestamp was either 0 or contrived from the start.